administrator/components/com_jedchecker/libraries/rules/language.php

```diff
@@ -0,0 +1,177 @@
+<?php
+/**
+ * @package    Joomla.JEDChecker
+ *
+ * @copyright  Copyright (C) 2021 Open Source Matters, Inc. All rights reserved.
+ *
+ * @license    GNU General Public License version 2 or later; see LICENSE.txt
+ */
+
+defined('_JEXEC') or die('Restricted access');
+
+
+// Include the rule base class
+require_once JPATH_COMPONENT_ADMINISTRATOR . '/models/rule.php';
+
+
+/**
+ * class JedcheckerRulesLanguage
+ *
+ * This class validates language ini file
+ *
+ * @since  2.3
+ */
+class JedcheckerRulesLanguage extends JEDcheckerRule
+{
+	/**
+	 * The formal ID of this rule. For example: SE1.
+	 *
+	 * @var    string
+	 */
+	protected $id = 'LANG';
+
+	/**
+	 * The title or caption of this rule.
+	 *
+	 * @var    string
+	 */
+	protected $title = 'COM_JEDCHECKER_LANG';
+
+	/**
+	 * The description of this rule.
+	 *
+	 * @var    string
+	 */
+	protected $description = 'COM_JEDCHECKER_LANG_DESC';
+
+	/**
+	 * Initiates the search and check
+	 *
+	 * @return    void
+	 */
+	public function check()
+	{
+		// Find all INI files of the extension (in the format tag.extension.ini or tag.extension.sys.ini)
+		$files = JFolder::files($this->basedir, '^[a-z]{2,3}-[A-Z]{2}\.\w+(?:\.sys)?\.ini$', true, true);
+
+		// Iterate through all the ini files
+		foreach ($files as $file)
+		{
+			// Try to validate the file
+			$this->find($file);
+		}
+	}
+
+	/**
+	 * Reads and validates an ini file
+	 *
+	 * @param   string  $file  - The path to the file
+	 *
+	 * @return bool True on success, otherwise False.
+	 */
+	protected function find($file)
+	{
+		$lines = file($file);
+		foreach ($lines as $lineno => $line)
+		{
+			$line = trim($line);
+
+			// Check for BOM sequence
+			if ($lineno === 0 && strncmp($line, "\xEF\xBB\xBF", 3) === 0)
+			{
+				$this->report->addError($file, JText::_('COM_JEDCHECKER_LANG_BOM_FOUND'), 1);
+				// Romeve BOM for further checks
+				$line = substr($line, 3);
+			}
+
+			// Skip empty lines, comments, and section names
+			if ($line === '' || $line[0] === ';' || $line[0] === '[')
+			{
+				continue;
+			}
+
+			// Report incorrect comment character
+			if ($line[0] === '#')
+			{
+				$this->report->addError($file, JText::_('COM_JEDCHECKER_LANG_INCORRECT_COMMENT') .
+					'<br>' . htmlspecialchars($line), $lineno);
+				continue;
+			}
+
+			// Check for "=" character in the line
+			if (strpos($line, '=') === false)
+			{
+				$this->report->addError($file, JText::_('COM_JEDCHECKER_LANG_WRONG_LINE') .
```

Comment:
Warning here..

Reply:
Maybe you are right (as partially incorrect language file doesn't prevent extension to be installed and used), but my logic was that most likely a line with missed "=" means there is a garbage in the language file (maybe it's not an INI format at all), and I don't see a case where it may be introduced for a reason.